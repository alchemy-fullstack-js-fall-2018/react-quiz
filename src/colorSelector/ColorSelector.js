import React from 'react';

export default class ColorSelector extends Component {

  state = {
    fontColor: 'red',
    backgroundColor: ''
  }

  onChange = ({ target }) => {
    // setState in async. It waits for all components to update
    this.setState({ [target.name]: target.value });
  };

  render() {
    const { fontColor, textArea, backgroundColor } = this.state;

    const fontOptions = [ 'black', 'white', 'blue', 'red', 'yellow'].map(font => {
      return <option key={font} value={font}>{font}</option>;
    });

    const backgroundColor = [ 'black', 'white', 'blue', 'red', 'yellow'].map(backColors => {
      return <option key={backColors} value={backColors}>{backColors}</option>;
    });


    const textStyles = () => {
      return {
        fontColor: fontColor,
        backgroundColor: backgroundColor
      }
    }

    return (
      <Fragment>
        <h1>Select Fonts</h1>

          <div>
            <label htmlFor="font">Select Font Color </label>
            <select name="font" defaultValue={font} onChange={this.onChange}>
              {fontOptions}
            </select>
          </div>

          <div>
            <label htmlFor="backColors">Select color background </label>
            <select name="backColors" defaultValue={backColors} onChange={this.onChange}>
              {backgroundColor}
            </select>
          </div>

          <div>
            <label htmlFor="textArea">Enter Text here</label>
            <input name="textArea" type="text" value={textArea} onChange={this.onChange} />
          </div>
          <div style={textStyles()}>
              {textArea}
            </div>
      </Fragment>
    );
}
