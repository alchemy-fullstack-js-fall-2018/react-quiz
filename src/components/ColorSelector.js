import React from 'react';

export default function ColorSelector({
  fontColor,
  backgroundColor,
  onChange
}) {
  return (
    <div>
      <label name="fontColor">Choose a font color:</label>
      <input
        name="fontColor"
        type="color"
        value={fontColor}
        onChange={onChange}
      />
      <label name="backgroundColor">Choose a background color:</label>
      <input
        name="backgroundColor"
        type="color"
        value={backgroundColor}
        onChange={onChange}
      />
    </div>
  );
}
