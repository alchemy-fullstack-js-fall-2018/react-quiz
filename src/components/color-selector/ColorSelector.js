import React, { Fragment } from 'react';

const ColorSelector = ({ textColor, backgroundColor, handleChange }) => {

  const textColorOptions = ['black', 'white', 'blue', 'red', 'yellow'].map(color => {
    return <option key={color} value={color}>{color}</option>;
  });
  const backgroundColorOptions = ['black', 'white', 'blue', 'red', 'yellow'].map(color => {
    return <option key={color} value={color}>{color}</option>;
  });
  return (
    <Fragment>
      <select name={textColor} defaultValue={textColor} onChange={handleChange}>
        {textColorOptions}
      </select>
      <select name={backgroundColor} defaultValue={backgroundColor} onChange={handleChange}>
        {backgroundColorOptions}
      </select>

    </Fragment>
  );
};

export default ColorSelector;
