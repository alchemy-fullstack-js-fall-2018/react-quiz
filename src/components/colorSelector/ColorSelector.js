import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

function ColorSelector({ artist, works }) {

  return (
    <Fragment>
      <input
        type="color"
      ></input>

    </Fragment>
  );
}

ColorSelector.propTypes = {
  // note: PropTypes.object.isRequired
};

export default ColorSelector;
