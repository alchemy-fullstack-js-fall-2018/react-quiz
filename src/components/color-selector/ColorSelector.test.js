import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('TextSelector Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ColorSelector location={{ pathname: '/somepath' }}/>);
  });
  it('renders a Songs component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
