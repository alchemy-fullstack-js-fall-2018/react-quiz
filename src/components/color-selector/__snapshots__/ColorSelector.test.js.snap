// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector Component renders a Songs component 1`] = `
<Fragment>
  <select>
    <option
      key="black"
      value="black"
    >
      black
    </option>
    <option
      key="white"
      value="white"
    >
      white
    </option>
    <option
      key="blue"
      value="blue"
    >
      blue
    </option>
    <option
      key="red"
      value="red"
    >
      red
    </option>
    <option
      key="yellow"
      value="yellow"
    >
      yellow
    </option>
  </select>
  <select>
    <option
      key="black"
      value="black"
    >
      black
    </option>
    <option
      key="white"
      value="white"
    >
      white
    </option>
    <option
      key="blue"
      value="blue"
    >
      blue
    </option>
    <option
      key="red"
      value="red"
    >
      red
    </option>
    <option
      key="yellow"
      value="yellow"
    >
      yellow
    </option>
  </select>
</Fragment>
`;
